sdk/src/trace/tracer.cc

```diff
@@ -8,12 +8,22 @@ namespace sdk
 {
 namespace trace
 {
+void Tracer::SetProcessor(std::shared_ptr<SpanProcessor> processor) noexcept
+{
+  std::atomic_store_explicit(&processor_, processor, std::memory_order_seq_cst);
+}
+
+SpanProcessor &Tracer::GetProcessor() const noexcept
+{
+  return *processor_;
+}
+
 nostd::unique_ptr<trace_api::Span> Tracer::StartSpan(
     nostd::string_view name,
     const trace_api::StartSpanOptions &options) noexcept
 {
-  return nostd::unique_ptr<trace_api::Span>{new (std::nothrow)
-                                                Span{this->shared_from_this(), name, options}};
+  return nostd::unique_ptr<trace_api::Span>{
+      new (std::nothrow) Span{this->shared_from_this(), processor_, name, options}};
```

Comment:
This needs an atomic_load to copy processor_

Reply:
@rnburn What would you suggest as the most elegant solution for gcc 4.8, which doesn't support atomic operations on shared pointers? Using `std::atomic` right away?

Reply:
Maybe wrap in an AtomicSharedPtr class and use preprocessor conditions to vary the behavior based on the version of c++. For gcc-4.8 and c++20 or later it can look like

```
AtomicSharedPtr {
 void store(const std::shared_ptr<SpanProcessor>& other) {
      ...
 }
std::shared_ptr<SpanProcessor> load() {
    ...
}
 private:
   std::atomic<std::shared_ptr<SpanProcessor>> ptr_;
}
```
for > gcc-4.8 and c++11 to c++17, it can look like
```
AtomicSharedPtr {
 void store(const std::shared_ptr<SpanProcessor>& other) {
      // call std::atomic_store
 }
std::shared_ptr<SpanProcessor> load() {
    // call std::atomic_load
}
 private:
   std::shared_ptr<SpanProcessor> ptr_;
}
```

That way it will use atomics when possible. `std::atomic<std::shared_ptr<T>>` would use a mutex for c++11 to c++17, but was specialized in c++20 to replace the std::atomic_store and std::atomic_load functions.